http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/dao/sam/SamApiClientProvider.scala

```diff
@@ -0,0 +1,51 @@
+package org.broadinstitute.dsde.workbench.leonardo.dao.sam
+
+import cats.effect.Async
+import cats.mtl.Ask
+import cats.syntax.all._
+import okhttp3.Protocol
+import org.broadinstitute.dsde.workbench.client.sam.ApiClient
+import org.broadinstitute.dsde.workbench.client.sam.api.{AzureApi, GoogleApi, ResourcesApi, UsersApi}
+import org.broadinstitute.dsde.workbench.leonardo.AppContext
+
+import scala.concurrent.duration._
+import scala.jdk.CollectionConverters._
+import scala.jdk.DurationConverters._
+
+/**
+ * Provides access to various Sam clients.
+ */
+trait SamApiClientProvider[F[_]] {
+  def resourcesApi(token: String)(implicit ev: Ask[F, AppContext]): F[ResourcesApi]
+  def usersApi(token: String)(implicit ev: Ask[F, AppContext]): F[UsersApi]
+  def googleApi(token: String)(implicit ev: Ask[F, AppContext]): F[GoogleApi]
+  def azureApi(token: String)(implicit ev: Ask[F, AppContext]): F[AzureApi]
+}
+
+class HttpSamApiClientProvider[F[_]](samUrl: String)(implicit F: Async[F]) extends SamApiClientProvider[F] {
+  private val okHttpClient = new ApiClient().getHttpClient
+  private val timeout = 30 seconds
+
+  private def getApiClient(token: String)(implicit ev: Ask[F, AppContext]): F[ApiClient] =
+    for {
+      ctx <- ev.ask
+      okHttpClientBuilder = okHttpClient.newBuilder
+        .readTimeout(timeout.toJava)
+        .protocols(Seq(Protocol.HTTP_1_1).asJava)
+      // TODO add otel interceptors
```

Comment:
is there a ticket for this to provide context?

Reply:
I'll make a tech debt ticket related to [IA-4295](https://broadworkbench.atlassian.net/browse/IA-4295)